client/src/pages/EnergiserDetail.js

```diff
@@ -29,7 +27,35 @@ function EnergiserDetail() {
             .catch((err) => {
                 console.error(err);
             });
-    }, [id]);
+
+        const favouriteStatus = localStorage.getItem(id);
```

Comment:
I'm a little bit confused about this, it looks like you're storing the same information in two different ways. You could easily derive the favourite status from the list of favouriteEnergizers - if it's in the list, then it's true and if it's not then it's false

Reply:
Thank u so much ,for your  time and your on point feedback .I will look into this .